keras_core/ops/core_test.py

```diff
@@ -247,3 +247,9 @@ def test_convert_to_tensor(self):
 
         with self.assertRaises(ValueError):
             ops.convert_to_numpy(KerasTensor((2,)))
+
+    def test_cond(self):
```

Comment:
Please also add a symbolic shape inference test case.

Reply:
To make the symbolic shape inference possible in all cases I had to implement a Cond operation to assert the `true_fn` and `false_fn` shapes to be equal. Hope that this isn't against the design philosophy